airflow/operators/docker_operator.py

```diff
@@ -238,9 +235,8 @@ def execute(self, context):
             if result['StatusCode'] != 0:
                 raise AirflowException('docker container failed: ' + repr(result))
 
-            if self.xcom_push_flag:
-                return self.cli.logs(container=self.container['Id']) \
-                    if self.xcom_all else str(line)
+            return self.cli.logs(container=self.container['Id']) \
+                if self.xcom_all else line.encode('utf-8')
```

Comment:
Hmmm. This is a somewhat expensive operation if the result is going to be thrown away at the next stage...